components/sync15/src/key_bundle.rs

```diff
@@ -75,53 +77,35 @@ impl KeyBundle {
         [base64::encode(&self.enc_key), base64::encode(&self.mac_key)]
     }
 
-    /// Returns the 32 byte digest by value since it's small enough to be passed
-    /// around cheaply, and easily convertable into a slice or vec if you want.
-    fn hmac(&self, ciphertext: &[u8]) -> Result<[u8; 32]> {
-        let mut out = [0u8; 32];
-        let key = PKey::hmac(self.hmac_key())?;
-        let mut signer = Signer::new(MessageDigest::sha256(), &key)?;
-        signer.update(ciphertext)?;
-        let size = signer.sign(&mut out)?;
-        // This isn't an Err since it really should not be possible.
-        assert!(
-            size == 32,
-            "Somehow the 256 bits from sha256 do not add up into 32 bytes..."
-        );
-        Ok(out)
+    fn hmac(&self, ciphertext: &[u8]) -> Result<Signature> {
+        let key = SigningKey::new(&digest::SHA256, self.hmac_key());
+        Ok(hmac::sign(&key, ciphertext)?)
```

Comment:
Excellent cleanup here.

As discussed in meeting, a future iteration on this code should aim to move the whole "check hmac and decrypt" machinery into a single `aead::open` call of some sort.  It will be complicated by sync needing to handle the `ciphertext` and `hmac` as two separate fields in the JSON, but I think it would be fine to require this code to concatenate them into a single buffer before calling `rc_crypto` to decrypt.